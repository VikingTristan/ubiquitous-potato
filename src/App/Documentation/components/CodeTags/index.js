import React from "react";
import { Link } from "react-router-dom";

import { ComponentPreview, DocContainer } from "@docutils";
import CodeTagsComponent from "@components/CodeTags";

const Overview = () => (
    <>
        <h2 id="overview">Overview</h2>
        <p>Code tags use Akkurat Mono as the font. Akkurat Mono can be found in <Link to="/docs/core/typography">Typography</Link></p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <p>{"\n"}
                This is an example of inline code <CodeTagsComponent type="primary" code="property.function()" /> that
                is presented using code tags.{"\n"}
            </p>
            <p>{"\n"}
                This is another example of inline code <CodeTagsComponent type="secondary" code="property.function()" /> that
                is presented using code tags.{"\n"}
            </p>
        </ComponentPreview>
    </>
);

const CodeTags = () => (
    <DocContainer docToc >
        <p className="lead">Code tags are used to present inline code.</p>
        <Overview />
    </DocContainer>
);

export default CodeTags;

/* for testing */
export { Overview };
