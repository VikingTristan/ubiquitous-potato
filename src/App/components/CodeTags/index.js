import React from "react";
import PropTypes from "prop-types";

const CodeTags = ({ type, code }) => (
    <span className={`code-tags code-tags-${type}`}>{code}</span>
);

CodeTags.propTypes = {
    type: PropTypes.string.isRequired,
    code: PropTypes.string.isRequired
};

export default CodeTags;
