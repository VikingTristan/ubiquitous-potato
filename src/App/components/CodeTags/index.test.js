import React from "react";
import { shallow } from "enzyme";

import CodeTags from "./index";

describe("Component: CodeTags", () => {
    it("is defined", () => {
        expect(CodeTags).toBeDefined();
    });

    it("renders with class code-tags-primary when primary is provided as type", () => {
        const wrapper = shallow(<CodeTags type="primary" code="test" />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.contains(<span className="code-tags code-tags-primary">test</span>)).toBeTruthy();
    });

    it("renders with class code-tags-secondary when secondary is provided as type", () => {
        const wrapper = shallow(<CodeTags type="secondary" code="test" />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.contains(<span className="code-tags code-tags-secondary">test</span>)).toBeTruthy();
    });

    it("prop type is marked as required", () => {
        console.error = jest.fn();

        const wrapper = shallow(<CodeTags code="test" />);

        expect(wrapper).toMatchSnapshot();
        expect(console.error).toHaveBeenCalled();
    });

    it("prop code is marked as required", () => {
        console.error = jest.fn();

        const wrapper = shallow(<CodeTags type="primary" />);

        expect(wrapper).toMatchSnapshot();
        expect(console.error).toHaveBeenCalled();
    });
});
