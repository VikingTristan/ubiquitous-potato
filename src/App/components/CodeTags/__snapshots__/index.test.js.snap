// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: CodeTags prop code is marked as required 1`] = `
<span
  className="code-tags code-tags-primary"
/>
`;

exports[`Component: CodeTags prop type is marked as required 1`] = `
<span
  className="code-tags code-tags-undefined"
>
  test
</span>
`;

exports[`Component: CodeTags renders with class code-tags-primary when primary is provided as type 1`] = `
<span
  className="code-tags code-tags-primary"
>
  test
</span>
`;

exports[`Component: CodeTags renders with class code-tags-secondary when secondary is provided as type 1`] = `
<span
  className="code-tags code-tags-secondary"
>
  test
</span>
`;
